### ISSUE TITLE
OpenDNS using MITM attacks wj32.org (forum & update servers)

### ISSUE THREAD

**Issue by dmex** [processhacker/processhacker#556]
_Originally posted by @nanoant on Dec 5, 2019 in https://github.com/processhacker/processhacker/issues/388#issuecomment-561833924_

![image](https://user-images.githubusercontent.com/103067/70235334-0fa69d00-1763-11ea-85cd-eb8447b9aa21.png)

**Comment by dmex** [comment]
![Screenshot_20200105-155236](https://user-images.githubusercontent.com/1306177/74129378-279b1100-4c2b-11ea-9b3e-b474feb9d7f4.png)
![Screenshot_20200105-155246](https://user-images.githubusercontent.com/1306177/74129388-2d90f200-4c2b-11ea-91be-7772b91deffc.png)
![Screenshot_20200210-113016](https://user-images.githubusercontent.com/1306177/74129394-3255a600-4c2b-11ea-81c5-e371fc629371.png)

**Comment by xcasxcursex** [comment]
Concerned I may have been pwned on this one :( I hope you'll spare a few minutes to help me confirm/deny it. TL;DR at the bottom, more info for posterity...

I don't login to the forum, but do follow the link from the readme.md here on github, to wj32.org, to get the nightlies from appveyor. (Might want to change that link?)

I get them fairly frequently, I'm unsure how long this has been going on, but I haven't seen any warnings (although my browser isn't checking every site visit with google so I wouldn't get one like the above) until today.

Today, I received an expired certificate warning, so after downloading (and intentionally not executing) the file I went to let you know the cert had expired, thinking it was just an oversight. Saw the forum thread that led me here.

I haven't actually needed to use the tool much recently, of course, Murphy's Law, the one exception I can recall was actually this morning.... During which I extracted the zip (I use the portable version), ran it, closed it, and attempted to delete it, and noticed it took a fair while to unload the kernel driver - not sure if that is relevant, but it did strike me as unusual. I had to wait about a minute between exiting the app and being able to delete that .sys file.

Now, I immediately deleted the one I just now downloaded - but I did run that other one this morning and it was downloaded on 13/02. Ran Windows Defender over it and it came up clean but.... I'm suitably nervous. I also have copies from back in January that I'm fairly sure I haven't run, and have been getting nightlies since like forever, so I'm not sure if I may have run one from further back that might have been a problem.


**TL;DR**
How long has this been going on?
Is there a file hash for legit appveyor CI builds that I can use to confirm/deny the copies I have are OK? 
Are there even any legit CI builds? / Does this even effect nightly CI builds? The download link from wj32.org is the same as the one I get from visiting appveyor, the file is coming from windows.net, but you say here the build servers are effected so perhaps just because it's coming from microsoft doesn't mean it's coming from microsoft, if you know what I mean.
Do we have any idea of what kind of threat this proposes? Are we talking about malware or a password leak that leaves no trace or what?

Any advice appreciated. I'd prefer to avoid a format if I can. Just got this machine running sweet............

**Comment by xcasxcursex** [comment]
Perhaps relevant, and probably worthy of a separate post - the expired cert I saw today, unlike your screenshots above, was issued by Let's Encrypt, and is dated from 18/11/2019 to 16/02/2020 - perhaps this actually is just a simple oversight and the legit cert really has expired?

**Comment by dmex** [comment]
> How long has this been going on?

I've been seeing that fake Cisco certificate from my mobile service intermittently since January.

> Are there even any legit CI builds?

Yes. You can view the official CI builds and download them direct from Appveyor. View the [history tab here](https://ci.appveyor.com/project/processhacker/processhacker/history) and select the build then click the [Artifacts tab](https://ci.appveyor.com/project/processhacker/processhacker/builds/30830003/artifacts) to download the binaries. 

> Does this even effect nightly CI builds? 

No. 

> The download link from wj32.org is the same as the one I get from visiting appveyor, the file is coming from windows.net, but you say here the build servers are effected

If you're downloading from windows.net then it's an official build. The nightly builds were previously hosted on wj32.org but I changed those locations back in December after Chrome and Edge started blocking downloads from wj32.org.

> Do we have any idea of what kind of threat this proposes? Are we talking about malware or a password leak that leaves no trace or what?

There are only two issues. Forum passwords and download links. The forum software (phpbb) transmits passwords in plain-text via HTTPS and a MITM attack can easily capture those passwords and the download links on the website could also be changed to different locations.

> Should we trust virus scanners saying we're clean, if we've run builds downloaded via the readme->wj32.org->appveyor? I'm assuming not...

Yes, if the builds were downloaded from appveyor then you're fine.

**Comment by dmex** [comment]
> the expired cert I saw today, unlike your screenshots above, was issued by Let's Encrypt, and is dated from 18/11/2019 to 16/02/2020 - perhaps this actually is just a simple oversight and the legit cert really has expired?

Let's Encrypt is the correct certificate. It's supposed to renew automatically but that didn't happen and we're not sure why. @wj32 needs to check what happened with the renewal and update the certificate to fix the security warnings.

**Comment by dmex** [comment]
Cisco Umbrella (OpenDNS) doesn't appear to be hijacking the connection anymore so I'll close this for now so it's not confused with the certificate expiration issue.

**Comment by xcasxcursex** [comment]
Thanks mate, I think I can stop panicking now. Appreciate your taking the time to put my mind at rest.

Best of luck with all this, sounds un-fun :(
And don't worry, this house has about 60 years of IT experience and we just talked about all the times we've seen certs expire because no reason, so you're not alone ;)

**Comment by MikeDiack** [comment]
Thanks gents for confirming this. I was a bit shocked about the certificate warnings yesterday (Sunday 16 Feb c. 9am UK time), that are still in effect. Thanks for clarifying what's going on. 
Are there any guesstimates (probably not), on when the dust will settle?

**Comment by dmex** [comment]
OpenDNS is using a MITM attack on the forum again... 
https://i.imgur.com/vXQEPzm.png
https://i.imgur.com/Nie8A46.png

**Comment by kdlslyv** [comment]
Not sure if related, but the Auto-Update prompted me for an update today, which windows defender quickly disposed as Trojan:Win32/Casdet!rfn.

**Comment by dmex** [comment]
@kdlslyv 

The two issues are not related. The updater is not susceptible to a MITM attack due to its use of asymmetric bcrypt signatures unlike the PKI chains (which are susceptible) used by websites and HTTPS. If the updater downloaded the update then it was an official nightly build but the installation was blocked by your antivirus.

Please use #564 or #454 for issues with antivirus.

**Comment by dmex** [comment]
Latest forged certificate issued by OpenDNS:
https://i.imgur.com/L3OEjMQ.png

I created a support ticket last week ([#683270](https://support.opendns.com/hc/en-us/requests/683270)) but still no response.

**Comment by dmex** [comment]
Latest forged certificate issued by OpenDNS:
https://i.imgur.com/8r5n3Pg.png

**Comment by xcasxcursex** [comment]
Is there anything we can do to help with this? You share a lot with us, it only seems right that we could make an effort to return the favour.

**Comment by dmex** [comment]
![image](https://user-images.githubusercontent.com/1306177/76681483-002ad000-663f-11ea-8e34-844f2c133549.png)

**Comment by dmex** [comment]
![image](https://user-images.githubusercontent.com/1306177/76713623-a4129980-676d-11ea-9123-e8c6663e6df9.png)

**Comment by xcasxcursex** [comment]
*facepalm*

Keep trying, they'll understand what's happening eventually.

**Comment by dmex** [comment]
Data breach notification as required by OAIC (Australia) and GPDR (Europe):
https://wj32.org/processhacker/forums/viewtopic.php?t=4019

Project website will be updated in the coming days.

**Comment by dmex** [comment]
> Keep trying, they'll understand what's happening eventually.

I tried but they closed the ticket after that last email (https://github.com/processhacker/processhacker/issues/556#issuecomment-599284891).

> Data breach notification as required by OAIC

OpenDNS refused to provide answers to relevant questions... I've submitted statements to the OAIC and Victoria Police today asking for a criminal investigation.

![image](https://user-images.githubusercontent.com/1306177/76818068-5a4eaf80-6850-11ea-9f21-cf08393c3e80.png)

**Comment by xcasxcursex** [comment]
Of course they did. Just wow.

I have a funny feeling they'll start to pay attention when a lawman says hello.

Good luck mate, hang in there.

**Comment by dmex** [comment]
> feeling they'll start to pay attention when a lawman says hello.

The Process Hacker organization is mandated by legislation to report security incidents to OAIC and provide formal statements - this was going to happen regardless of what OpenDNS provided the development team.

> Good luck mate, hang in there.

State and federal police here in Australia are now officially investigating. We'll get some answers about what has happened and hopefully some criminal prosecutions for the individuals whom carried out the attack.

**Comment by linuxandria** [comment]
My college is blocking the download site and it cites malware as the reason (they appear to use OpenDNS)

**Comment by xcasxcursex** [comment]
Just popped by to see how this was fairing, and had to scroll a while to find it. Perhaps this issue should be pinned since it's potentially an attack vector to end users?

**Comment by dbratanis** [comment]
Currently using latest version of the app. Bitdefender is reporting and blocking the app because of phishing attempts.

https://imgur.com/a/yfZyqAE

Using Quad9 as main dns. 

I dont know if this is the best place to report this but bitdefender now allowing to use the app if not place in exception mode.

**Comment by Biswa96** [comment]
@linuxandria If your college blocks anything the first choice would be to contact them. Otherwise you can download the binaries from Appveyor artifacts. See the commits of the repository and click on green tick on any lates commit. Otherwise you can fork this repository and setup your own CI system.

@dbratanis Ya, ProcessHacker now-a-days becomes a popular one among antivirus program, there is a mega thread about it.

**Comment by dbratanis** [comment]
> @dbratanis Ya, ProcessHacker now-a-days becomes a popular one among antivirus program, there is a mega thread about it.

My AV if you look the image is blocking this app because of the phishing attempt on wj32.org not because it containing any malware. I use the app long time with BD with no problem and found this topic for wj32 and thought to report it.

**Comment by linuxandria** [comment]
> @Biswa96 
> > @dbratanis Ya, ProcessHacker now-a-days becomes a popular one among antivirus program, there is a mega thread about it.
> 
> My AV if you look the image is blocking this app because of the phishing attempt on wj32.org not because it containing any malware. I use the app long time with BD with no problem and found this topic for wj32 and thought to report it. 
> 

Sophos throws a fit about a couple dlls in process hacker here

**Comment by linuxandria** [comment]
> @linuxandria If your college blocks anything the first choice would be to contact them. Otherwise you can download the binaries from Appveyor artifacts. See the commits of the repository and click on green tick on any lates commit. Otherwise you can fork this repository and setup your own CI system.
> 
> @dbratanis Ya, ProcessHacker now-a-days becomes a popular one among antivirus program, there is a mega thread about it.

And what do you think the chances of them unblocking because I asked :P

Anyhow downloaded over mobile data

**Comment by dmex** [comment]
> ProcessHacker now-a-days becomes a popular one among antivirus program

Antivirus programs are the modern day version of snake oil salesmen pushing an elixir to cure your illness. I've worked in IT for 16 years and seen a lot of changes but the one constant is that not once have I ever seen antivirus software stop an attacker.

> I dont know if this is the best place to report this but bitdefender now allowing

@dbratanis Can you check if Bitdefender is still blocking the domain? I submitted a false positive a few days ago and they should have updated.

> what do you think the chances of them unblocking because I asked

@linuxandria You should always ask companies about these types of actions and request changes. If they feel like not providing service then switch products to somewhere that does.

**Comment by dbratanis** [comment]
@dmex 
> @dbratanis Can you check if Bitdefender is still blocking the domain? I submitted a false positive a few days ago and they should have updated.


Hello i did a fresh installation, clicked _check for updates_ went to website, no pop ups or warnings from Bitdefender. Seems everything is back to normal.

**Comment by xcasxcursex** [comment]
Closed? What's happening with this?
